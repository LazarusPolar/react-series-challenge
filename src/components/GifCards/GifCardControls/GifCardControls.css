.GifCardControls {
    z-index: 10px;
    display: inline-block;
}

.GifCardControls button {
    width: 400px;
    height: 250px;
    z-index: 10px;
    display: inline-block;
}