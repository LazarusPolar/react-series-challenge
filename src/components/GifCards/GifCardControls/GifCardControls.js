import React from 'react';

const GifCardControls = (props) => {
    //console.log("props", props);

    return(
        <div className = "GifCardControls">
            <button onClick = {props.addRemoveGif}> {props.isFavorite ? <h1>Quitar Favorito</h1> : <h1>Poner Favorito</h1>} </button>
        </div>
    );
}

export default GifCardControls;