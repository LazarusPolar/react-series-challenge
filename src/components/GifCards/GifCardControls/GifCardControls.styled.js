import styled from 'styled-components';

export const StyledGifCardControls = styled.div`
    padding-right: 0px;
    padding-left: 0px;
    margin: 0px;
    border: 0px;
`

export const StyledButton = styled.button`
    padding-right: 20px;
    padding-left: 20px;
    margin: 0px;
    border: 0px;

    &.active {
        color: red;
    }

    :hover {
        background-color: #00ff00;
    }

    h1 {
        color: #000000;
        font-family: 'Quicksand', sans-serif;  
    }

`


