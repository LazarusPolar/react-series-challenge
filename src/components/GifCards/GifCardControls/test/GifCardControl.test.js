import GifCardControls from '../GifCardControls';
import React from 'react';
import { StyledGifCardControls, StyledButton } from '../GifCardControls.styled'
import { shallow } from 'enzyme';

describe('GifCardControls', () => {
  const func = jest.fn();
  const valueKey = 1;
  const favorite = false;
  let wrapper;

    beforeEach(() => {
        wrapper = shallow(
            <GifCardControls
            key = {valueKey} 
            addRemoveGif = {func} 
            isFavorite = {favorite} />
        );
    });

    test('should render', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should set GifCardControls ', () => {
        const comp = shallow(
        <StyledGifCardControls>
            <StyledButton active onClick = {func}> 
                {favorite ? <h1>ðŸ˜¢ Remove me ðŸ˜¢</h1> : <h1>âœ¨ Fav me! âœ¨</h1>} 
            </StyledButton>
        </StyledGifCardControls>)

        expect(comp).toHaveLength(1);
    });

})
