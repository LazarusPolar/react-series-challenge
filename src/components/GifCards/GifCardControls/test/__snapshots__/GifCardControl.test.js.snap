// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifCardControls should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GifCardControls
    addRemoveGif={[MockFunction]}
    isFavorite={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <styled.button
        active={true}
        onClick={[MockFunction]}
      >
         
        <h1>
          ✨ Fav me! ✨
        </h1>
         
      </styled.button>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "active": true,
        "children": Array [
          " ",
          <h1>
            ✨ Fav me! ✨
          </h1>,
          " ",
        ],
        "onClick": [MockFunction],
      },
      "ref": null,
      "rendered": Array [
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "✨ Fav me! ✨",
          },
          "ref": null,
          "rendered": "✨ Fav me! ✨",
          "type": "h1",
        },
        " ",
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <styled.button
          active={true}
          onClick={[MockFunction]}
        >
           
          <h1>
            ✨ Fav me! ✨
          </h1>
           
        </styled.button>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": true,
          "children": Array [
            " ",
            <h1>
              ✨ Fav me! ✨
            </h1>,
            " ",
          ],
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": Array [
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "✨ Fav me! ✨",
            },
            "ref": null,
            "rendered": "✨ Fav me! ✨",
            "type": "h1",
          },
          " ",
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
